commons-numbers-examples/examples-jmh/src/main/java/org/apache/commons/numbers/examples/jmh/gamma/ErfPerformance.java

```diff
@@ -0,0 +1,720 @@
+            final String impl = getImplementation();
+            if (IMP_NUMBERS_1_0.equals(impl)) {
+                return ErfPerformance::inverseErfc;
+            } else if (IMP_NUMBERS_1_1.equals(impl)) {
+                return InverseErfc::value;
+            } else {
+                throw new IllegalStateException(UNKNOWN + impl);
+            }
+        }
+
+        /** {@inheritDoc} */
+        @Override
+        protected void verify() {
+            final DoubleUnaryOperator function = getFunction();
+            final double relativeEps = 1e-12;
+            for (final double x : getNumbers()) {
+                assert 0 <= x && x <= 2 : ERFC_DOMAIN_ERROR + x;
+
+                // Implementations may not compute a round-trip
+                // to a suitable accuracy as:
+                // x -> 0 : t -> big
+                // |x| -> 1 : t -> 0
+                // x -> 2 : t -> -big
+                if (x < 1e-10 || Math.abs(x - 1) < 1e-10 || x > 2 - 1e-10) {
+                    continue;
+                }
+                final double t = function.applyAsDouble(x);
+                assertEquals(x, Erfc.value(t), Math.abs(x) * relativeEps,
+                    () -> getImplementation() + " erfc " + t);
+            }
+        }
+    }
+
+    /**
+     * Make a signed double in the range [-1, 1).
+     *
+     * @param rng Random generator
+     * @return u in [-1, 1)
+     */
+    private static double makeSignedDouble(SplittableRandom rng) {
+        // As per o.a.c.rng.core.utils.NumberFactory.makeDouble(long) but using a signed
+        // shift of 10 in place of an unsigned shift of 11.
+        // Use the upper 54 bits on the assumption they are more random.
+        // The sign bit is maintained by the signed shift.
+        // The next 53 bits generates a magnitude in the range [0, 2^53) or [-2^53, 0).
+        return (rng.nextLong() >> 10) * 0x1.0p-53;
+    }
+
+    /**
+     * Returns the inverse complementary error function.
+     *
+     * <p>This is the implementation in Commons Numbers 1.0.
+     *
+     * @param x Value in [0, 2].
+     * @return t such that {@code x = erfc(t)}
+     */
+    private static double inverseErfc(final double x) {
+        return inverseErf(1 - x);
+    }
+
+    /**
+     * Returns the inverse error function.
+     *
+     * <p>This implementation is described in the paper:
+     * <a href="http://people.maths.ox.ac.uk/gilesm/files/gems_erfinv.pdf">Approximating
+     * the erfinv function</a> by Mike Giles, Oxford-Man Institute of Quantitative Finance,
+     * which was published in GPU Computing Gems, volume 2, 2010.
+     * The source code is available <a href="http://gpucomputing.net/?q=node/1828">here</a>.
+     * </p>
+     *
+     * <p>This is the implementation in Commons Numbers 1.0.
+     *
+     * @param x Value in [-1, 1].
+     * @return t such that {@code x = erf(t)}
+     */
+    private static double inverseErf(final double x) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 71 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.